### PR TITLE
Updated ABI definitions & parsing

### LINKED ISSUE #1549: Contract deploy method should support Vec<u8>
I get an error `main.749f3709.js:21450 Uncaught Error: createType({ "elems": "Vec" }):: Unable to find plain type for {"info":6,"type":"Vec"}`, when I call deploy at https://polkadot.js.org/apps/#/contracts/code.
sALNPfuHKHgwCxEqXIRgdfDM
 When I remove Vec<u8> in deploy, the contract can be instantiated normally.

### LINKED ISSUE #1660: Support adjusted contracts ABI
https://github.com/paritytech/ink/pull/296

This opens the door to get rid of the custom parser and just define the objects and parse directly into them.

### PR DESCRIPTION
**Pull request by jacogr**
- Closes #1660 Closes #1549 
- Implement types definitions for https://github.com/paritytech/ink/pull/296
- Implement type definitions for https://github.com/type-metadata/type-metadata/pull/36

### CODE REVIEW

File: packages/types/src/interfaces/contractsAbi/definitions.ts [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+const registry = {
+  MtClikeEnumVariant: {
+    name: 'u32',
+    discriminant: 'u64'
+  },
+  MtEnumVariant: {
+    _enum: {
+      Unit: 'MtEnumVariantUnit',
+      Struct: 'MtEnumVariantStruct',
+      // #[serde(rename = "tupleStruct")]
```
Comment (jacogr) [inline packages/types/src/interfaces/contractsAbi/definitions.ts]: All these comments left in, could be problematic since it is non-standard.

**Comment by ascjones** [comment]
@jacogr would you mind merging master again? Hopefully this will be the last time

**Comment by jacogr** [comment]
Just merged it as you were typing that ... :) Great mind think alike, fools never differ...

**Comment by ascjones** [comment]
> @jacogr would you mind merging master again? Hopefully this will be the last time

Er yeah about that, one more time please :sweat_smile:

**Comment by jacogr** [comment]
Merged, sorry about the delay here.

**Pull request merged**